enhancements/installer/inputs-from-stdin.md

```diff
@@ -0,0 +1,182 @@
+
+This OpenShift enhancement proposes to allow the installer to read a stream from STDIN on the main targets that consume inputs.
+
+
+## Motivation
+
+Infrastructure-as-Code (e.g. GitOps) is an operational best practice for provisioning and managing large numbers of clusters and services. A typical IaC workflow involves pulling Kubernetes resource manifest templates of a specific version from a `git` repo, rendering those templates for a given cluster instance, and then running `openshift-install` (resp. `oc apply`) on the rendered manifests to install (resp. update) a cluster.
+
+Pulling and rendering manifest templates is conveniently done via `kustomize` - which is particularly suited for an IaC workflow - but other tools like `helm` and `jsonnet` have their followers, too. These tools have in common that they render templates into a stream of manifests sent to `stdout`.
+
+As `openshift-install` currently only supports reading manifest files from the file system and requires files to follow a specific naming and directory structure convention, IaC workflows become cumbersome and often involve users writing their own Bash scripts or Ansible playbooks to automate the workflows. By giving the installer the ability to read from `stdin`, we will enable a simplified workflow pattern via pipelines while still being capable of scaling with the complexity of users’ needs.
+
+### Goals
+
+Enable the Openshift Installer to support pipelined workflows by reading inputs from STDIN as an alternative to reading from the filesystem.
+
+### Non-Goals
+
+- Change the existing workflow to deploy an OpenShift cluster.
+- Couple the Openshift Installer with Kustomize or other manifest tooling
+
+
+## Proposal
+
+Modify the OpenShift Installer to accept inputs from STDIN for all targets. For example: 
+
+- Create manifests from customized install-config.yaml: `kustomize build github.com/myorg/myBlueprint/install-config-kustomization | openshift-install create manifests -f -`
+- Create ignition files from customized set of manifests: `kustomize build ~/myBlueprint/01_manifests | openshift-install create ignition-configs -f -`
+- Create cluster from customized set of manifests: `kustomize build ~/myBlueprint/01_manifests | openshift-install create cluster -f -`
+
+The syntax and workflow will be similar to `oc apply -f -`.
+
+A straight example of a use case is explained next. From Kustomize v3.7.0 it's possible to create reusable customizations called `Components`. Acording to this enhancement proposal the output of `openshift-install --kustomization create manifests` will generate the following structure in the chosen directory:
```

Comment:
Thanks for pointing this out. We added this as response to comment https://github.com/openshift/enhancements/pull/575#discussion_r555414105 to add a global flag to enable/disable the behavior.

We need to further clarify this and should probably not call the flag `--kustomization` for the reason I stated in https://github.com/openshift/enhancements/pull/575#discussion_r581813012 but something less opinionated.

Reply:
The `--kustomization` is not right here. `--stdin` would be a better fit, since we don't want to be opinionated on the tool that the user must leverage to perform customizations on installer inputs.